src/components/captions-menu/captions-menu.tsx

```diff
@@ -58,6 +59,9 @@ class CaptionsMenu extends Component<any, any> {
       return;
     }
     this.props.player.selectTrack(textTrack);
+    // Update the Redux state to reflect the current captions state
+    const isCaptionsEnabled = typeof textTrack === 'object' && textTrack.language !== 'off';
```

Comment:
use the const 
https://github.com/kaltura/playkit-js-ui/blob/704c0304c6cef949944cd449b85a8969315153df/src/reducers/settings.ts#L4